File: mainActivity/src/main/java/net/airvantage/utils/IAirVantageClient.java
package net.airvantage.utils;

import java.io.IOException;
import java.util.List;

import net.airvantage.model.AirVantageException;
import net.airvantage.model.alert.v1.AlertRule;
import net.airvantage.model.Application;
import net.airvantage.model.ApplicationData;
import net.airvantage.model.AvSystem;
import net.airvantage.model.Protocol;
import net.airvantage.model.User;
import net.airvantage.model.UserRights;

public interface IAirVantageClient {

    void createAlertRule(AlertRule alertRule, String application) throws IOException, AirVantageException;

    AlertRule getAlertRuleByName(String name, String application) throws IOException, AirVantageException;

    void setApplicationData(String applicationUid, List<ApplicationData> data) throws IOException, AirVantageException;

    List<Application> getApplications(String appType) throws IOException, AirVantageException;

    Application createApplication(Application application) throws IOException, AirVantageException;

    void setApplicationCommunication(String applicationUid, List<Protocol> protocols) throws IOException, AirVantageException;

    void updateSystem(AvSystem system) throws IOException, AirVantageException;

    void logout() throws IOException, AirVantageException;

    UserRights getUserRights() throws IOException, AirVantageException;

    User getCurrentUser() throws IOException, AirVantageException;

}

File: mainActivity/src/main/java/net/airvantage/utils/IAirVantageClient.kt
package net.airvantage.utils

import java.io.IOException

import net.airvantage.model.AirVantageException
import net.airvantage.model.alert.v1.AlertRule
import net.airvantage.model.Application
import net.airvantage.model.ApplicationData
import net.airvantage.model.AvSystem
import net.airvantage.model.Protocol
import net.airvantage.model.User
import net.airvantage.model.UserRights

interface IAirVantageClient {

    val userRights: UserRights

    val currentUser: User

    @Throws(IOException::class, AirVantageException::class)
    fun createAlertRule(alertRule: AlertRule, application: String, system: AvSystem)

    @Throws(IOException::class, AirVantageException::class)
    fun updateAlertRule(alertRule: AlertRule, application: String, system: AvSystem)

    @Throws(IOException::class, AirVantageException::class)
    fun getAlertRuleByName(name: String, system: AvSystem): AlertRule?

    @Throws(IOException::class, AirVantageException::class)
    fun deleteAlertRule(alertRule: AlertRule)

    @Throws(IOException::class, AirVantageException::class)
    fun setApplicationData(applicationUid: String, data: List<ApplicationData>)

    @Throws(IOException::class, AirVantageException::class)
    fun getApplications(appType: String): List<Application>?

    @Throws(IOException::class, AirVantageException::class)
    fun createApplication(application: Application): Application

    @Throws(IOException::class, AirVantageException::class)
    fun setApplicationCommunication(applicationUid: String, protocols: List<Protocol>)

    @Throws(IOException::class, AirVantageException::class)
    fun updateSystem(system: AvSystem)

    @Throws(IOException::class, AirVantageException::class)
    fun logout()
}
